src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -1,14 +1,168 @@
 package cromwell.binding
 
+import cromwell.binding.AstTools.AstNodeName
+import cromwell.parser.WdlParser.{AstList, Ast, AstNode}
+import scala.collection.JavaConverters._
+
+
+object Scope {
+
+  /**
+   * Utility class to generate a ScopeTree from an Ast.
+   */
+  private class ScopeTreeGenerator {
+
+    /* Keep track of the # of created Scatter nodes. */
+    var scatterCount: Int = 0
+
+  /*
+  * Retrieve the list of AstNodes from the AST body.
+  * Return empty seq if body is null or is not a list.
+  */
+    private def getChildrenList(ast: Ast): Seq[AstNode] = {
+      val body: AstNode = ast.getAttribute("body")
+      if(body == null) Seq.empty
+      else {
+        body match {
+          case list: AstList => list.asScala.toVector
+          case _ => Seq.empty
+        }
+      }
```

Comment:
Or I suppose I could have scrolled down a bit and seen that @mcovarr already covered it :)